robusta_krr/core/runner.py

```diff
@@ -141,11 +144,11 @@ def _format_result(self, result: RunResult) -> RunResult:
             for resource, recommendation in result.items()
         }
 
-    async def _calculate_object_recommendations(self, object: K8sObjectData) -> RunResult:
+    async def _calculate_object_recommendations(self, object: K8sObjectData) -> Optional[RunResult]:
         prometheus_loader = self._get_prometheus_loader(object.cluster)
 
         if prometheus_loader is None:
-            return {resource: ResourceRecommendation.undefined("Prometheus not found") for resource in ResourceType}
+            return None
```

Comment:
how does this change the output yaml?

Reply:
So this is a breaking change, but previously not found prometheus resulted in a success scan in UI (and that was wrong and confusing)